File: iosApp/iosApp/Theme/Theme.swift
import SwiftUI

// MARK: - Dragon Ball Theme Colors (matching Android exactly)

extension Color {
    // Primary Colors - matching Android hex values
    // DragonBallOrange = 0xFFFF6B35
    static let dragonBallOrange = Color(red: 255/255, green: 107/255, blue: 53/255)
    // DragonBallBlue = 0xFF2A9DF4
    static let dragonBallBlue = Color(red: 42/255, green: 157/255, blue: 244/255)
    // DragonBallYellow = 0xFFFFD23F
    static let dragonBallYellow = Color(red: 255/255, green: 210/255, blue: 63/255)

    // Additional colors from Android
    // DragonBallDarkOrange = 0xFFD94E2A
    static let dragonBallDarkOrange = Color(red: 217/255, green: 78/255, blue: 42/255)
    // DragonBallLightOrange = 0xFFFFB084
    static let dragonBallLightOrange = Color(red: 255/255, green: 176/255, blue: 132/255)
    // DragonBallPurple = 0xFF6A4C93
    static let dragonBallPurple = Color(red: 106/255, green: 76/255, blue: 147/255)
    // DragonBallDarkBlue = 0xFF1E3A8A
    static let dragonBallDarkBlue = Color(red: 30/255, green: 58/255, blue: 138/255)
    // DragonBallGold = 0xFFFFA500
    static let dragonBallGold = Color(red: 255/255, green: 165/255, blue: 0/255)

    // Background & Surface - matching Android light theme
    // background = 0xFFFFF8F0
    static let appBackground = Color(red: 255/255, green: 248/255, blue: 240/255)
    // surface = White
    static let appSurface = Color.white
    // surfaceVariant = 0xFFFFF4E8
    static let appSurfaceVariant = Color(red: 255/255, green: 244/255, blue: 232/255)
}

// MARK: - Navigation Bar Appearance

extension View {
    func setupNavigationBarAppearance() -> some View {
        self.onAppear {
            let appearance = UINavigationBarAppearance()
            appearance.configureWithOpaqueBackground()
            appearance.backgroundColor = UIColor(red: 255/255, green: 248/255, blue: 240/255, alpha: 1)
            appearance.titleTextAttributes = [.foregroundColor: UIColor(red: 255/255, green: 107/255, blue: 53/255, alpha: 1)]
            appearance.largeTitleTextAttributes = [.foregroundColor: UIColor(red: 255/255, green: 107/255, blue: 53/255, alpha: 1)]

            // Back button color
            let buttonAppearance = UIBarButtonItemAppearance()
            buttonAppearance.normal.titleTextAttributes = [.foregroundColor: UIColor(red: 255/255, green: 107/255, blue: 53/255, alpha: 1)]
            appearance.buttonAppearance = buttonAppearance
            appearance.backButtonAppearance = buttonAppearance

            UINavigationBar.appearance().standardAppearance = appearance
            UINavigationBar.appearance().scrollEdgeAppearance = appearance
            UINavigationBar.appearance().compactAppearance = appearance
            UINavigationBar.appearance().tintColor = UIColor(red: 255/255, green: 107/255, blue: 53/255, alpha: 1)
        }
    }
}

File: androidApp/src/main/kotlin/com/example/kdragonball/android/ui/theme/Theme.kt
package com.example.kdragonball.android.ui.theme

import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.darkColorScheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color

// Dragon Ball themed colors
val DragonBallOrange = Color(0xFFFF6B35)
val DragonBallBlue = Color(0xFF2A9DF4)
val DragonBallYellow = Color(0xFFFFD23F)
val DragonBallDarkOrange = Color(0xFFD94E2A)
val DragonBallLightOrange = Color(0xFFFFB084)
val DragonBallPurple = Color(0xFF6A4C93)
val DragonBallDarkBlue = Color(0xFF1E3A8A)
val DragonBallGold = Color(0xFFFFA500)

private val LightColorScheme =
    lightColorScheme(
        primary = DragonBallOrange,
        onPrimary = Color.White,
        primaryContainer = DragonBallLightOrange,
        onPrimaryContainer = DragonBallDarkOrange,
        secondary = DragonBallBlue,
        onSecondary = Color.White,
        secondaryContainer = Color(0xFFB3E5FC),
        onSecondaryContainer = DragonBallDarkBlue,
        tertiary = DragonBallYellow,
        onTertiary = Color.Black,
        background = Color(0xFFFFF8F0),
        onBackground = Color(0xFF2D2D2D),
        surface = Color.White,
        onSurface = Color(0xFF2D2D2D),
        surfaceVariant = Color(0xFFFFF4E8),
        onSurfaceVariant = Color(0xFF5D5D5D),
        error = Color(0xFFD32F2F),
        onError = Color.White
    )

private val DarkColorScheme =
    darkColorScheme(
        primary = DragonBallOrange,
        onPrimary = Color.Black,
        primaryContainer = DragonBallDarkOrange,
        onPrimaryContainer = DragonBallLightOrange,
        secondary = DragonBallBlue,
        onSecondary = Color.Black,
        secondaryContainer = DragonBallDarkBlue,
        onSecondaryContainer = Color(0xFFB3E5FC),
        tertiary = DragonBallYellow,
        onTertiary = Color.Black,
        background = Color(0xFF1A1A1A),
        onBackground = Color(0xFFE8E8E8),
        surface = Color(0xFF2D2D2D),
        onSurface = Color(0xFFE8E8E8),
        surfaceVariant = Color(0xFF3D3D3D),
        onSurfaceVariant = Color(0xFFCCCCCC),
        error = Color(0xFFEF5350),
        onError = Color.Black
    )

@Composable
fun AppTheme(darkTheme: Boolean = false, content: @Composable () -> Unit) {
    val colorScheme = if (darkTheme) DarkColorScheme else LightColorScheme

    MaterialTheme(
        colorScheme = colorScheme,
        content = content
    )
}
